polaris/hub/polarisfs.py

```diff
@@ -87,16 +94,22 @@ def ls(
         if timeout is None:
             timeout = self.default_timeout
 
+        cached_listings = self._ls_from_cache(path)
+        if cached_listings is not None:
+            return cached_listings if detail else [d["name"] for d in cached_listings]
+
+        cached_listings = self._ls_from_cache(self._parent(path))
+        if cached_listings is not None:
+            self.dircache[path] = []
+            return cached_listings if detail else [d["name"] for d in cached_listings]
```

Comment:
Why do we check ls for both the parent and the path? I feel like we should only do so for the path, as is e.g. done in [this existing implementation](https://github.com/fsspec/filesystem_spec/blob/518984d494e48f007d8aa48de882ed7014da2574/fsspec/implementations/dbfs.py#L72).